kratos/python_scripts/time_based_ascii_file_writer_utility.py

```diff
@@ -126,43 +131,17 @@ def __ValidateAndAssignOutputFolderPath(self):
         if self.file_name == "":
             raise Exception('No "file_name" was specified!')
         # check and correct file extension
-        if not self.file_name.endswith(".dat"):
-            self.file_name += ".dat"
+        if self.file_name.suffix != ".dat":
+            self.file_name = self.file_name.with_suffix(".dat")
 
-        # check if relative path was erroneously specified in file name
         raw_path, raw_file_name = os.path.split(self.file_name)
+        if not raw_path == '':
+            warn_msg  = "Path contained wrongly in file_name "+ self.file_name +" is being ignored.\n"
+            warn_msg += "Use parameter output_path to specify it correctly."
+            KratosMultiphysics.Logger.PrintWarning("TimeBasedAsciiFileWriterUtility", warn_msg)
```

Comment:
```suggestion
        if not raw_path == '':
            warn_msg  = "Path contained wrongly in file_name "+ self.file_name +" is being ignored.\n"
            warn_msg += "Use parameter output_path to specify it correctly."
            KratosMultiphysics.Logger.PrintWarning("TimeBasedAsciiFileWriterUtility", warn_msg)
            raw_file_name = os.path.join(raw_path, raw_file_name)
```

this should ensure the backwards compatibility and hopefully also fix the test